hdbscan/validity.py

```diff
@@ -104,13 +114,19 @@ def all_points_mutual_reachability(X, labels, cluster_id,
                                              **kwd_args)
         d = X.shape[1]
 
-    core_distances = all_points_core_distance(distance_matrix.copy(), d=d)
-    core_dist_matrix = np.tile(core_distances, (core_distances.shape[0], 1))
+    if no_coredist:
+        return distance_matrix, None
+
+    else:
+        core_distances = all_points_core_distance(distance_matrix.copy(), d=d)
+        core_dist_matrix = np.tile(core_distances, (core_distances.shape[0], 1))
+        stacked_distances = np.dstack(
+            [distance_matrix, core_dist_matrix, core_dist_matrix.T])
 
-    result = np.dstack(
-        [distance_matrix, core_dist_matrix, core_dist_matrix.T]).max(axis=-1)
+        if print_max_euclid_to_coredist_ratios:
+            print_max_ratio(stacked_distances)
```

Comment:
As a general rule printing things unless explicitly asked, usually with a ``verbose=True`` or similar, is frowned upon. You have a guard statement on this, which is good, but in practice it might be beneficial to unify all of this at the top call level with a ``verbose`` option. See below...

Reply:
Thanks for the feedback! I applied your suggestions. If you don't post a further message to this thread, I will assume that you are fine with the messages invariably being emitted via stdout (if the user opts-in via the `verbose` kwarg).
  
(As opposed to using a logger as mentioned here: https://github.com/scikit-learn-contrib/hdbscan/pull/552#issuecomment-1193848434)